source/MaterialXRuntime/RtNodeGraph.h

```diff
@@ -31,12 +31,18 @@ class RtNodeGraph : public RtNode
     /// Remove an input attribute from the graph.
     void removeInput(const RtToken& name);
 
+    /// Rename an input attribute in the graph.
+    void renameInput(const RtToken& name, const RtToken& newName);
+
     /// Add an output attribute to the graph.
     RtOutput createOutput(const RtToken& name, const RtToken& type, uint32_t flags = 0);
 
     /// Remove an output attribute from the graph.
     void removeOutput(const RtToken& name);
 
+    /// Rename an output attribute in the graph.
+    void renameOutput(const RtToken& name, const RtToken& newName);
```

Comment:
In public API only expose rename for inputs and outputs on NodeGraphs for now, which we know we need. 
(Renaming attributes on NodeDefs and Nodes has more complications so we'll wait until/if we need it).